jest 사용 세팅 및 실행 시 에러 이슈

# 세팅 과정

1. 관련 패키지 설치 (jest는 이미 설치되어 있었음)
```
pnpm install -D @types/jest ts-jest ts-node
```
2. jestconfig.cjs 파일 생성

``` jsx
module.exports = {
    preset : 'ts-jest',
    testEnvironment : 'node',
};
```

# 에러

```
Cannot find module 'src/constants' from 'src/config/defaultConfig.ts'

    Require stack:
      src/config/defaultConfig.ts
      src/commands/insertHeader/insertHeader.ts
      src/commands/insertHeader/insertHeader.test.ts

    > 1 | import { DEFAULT_COMMIT_TYPES, DEFAULT_EMOJIS, DEFAULT_REPLACE_EXAMPLES } from 'src/constants';
```

위와 같이
defaultConfig.ts 파일에서 
import 경로를 

```
'src/constants'
```
와 같이 쓴 부분에서 에러가 나는데요
이 부분을

```
'./../constants';
```
와 같이 고치면 에러가 해결되고 jest가 잘 실행됨을 확인 했습니다.
@ninthsun91 이 부분이 수정되어야 할 것 같은데 다른 이슈가 있을까요?

--- a/package.json
+++ b/package.json
@@ -47,6 +47,7 @@
     "husky": "^8.0.0",
     "jest": "^29.6.2",
     "lint-staged": "^13.2.3",
+    "ts-jest": "^29.1.1",
     "ts-node": "^10.9.1",
     "ts-patch": "^3.0.2",
     "typescript": "^5.1.6",

--- a/package.json
+++ b/package.json
@@ -16,7 +16,7 @@
     "start": "pnpm build && node --loader=./dist/config/loader.js --experimental-specifier-resolution=node --no-warnings dist/index.js",
     "lint": "eslint './src/**/*.{ts,tsx,js,jsx}'",
     "lint:fix": "eslint --fix './src/**/*.{ts,tsx,js,jsx}'",
-    "test": "echo \"Error: no test specified\" && exit 1",
+    "test": "jest ./test",
     "prepare": "husky install",
     "postinstall": "ts-patch install"
   },
@@ -55,5 +55,18 @@
   },
   "lint-staged": {
     "*.{ts,tsx,js,jsx}": "eslint --cache --fix"
+  },
+  "jest": {
+    "moduleDirectories": ["node_modules", "src"],
+    "moduleFileExtensions": ["js", "json", "ts"],
+    "moduleNameMapper": {
+      "src/(.*)$": "<rootDir>/src/$1"
+    },
+    "rootDir": ".",
+    "testEnvironment": "node",
+    "testRegex": ".*\\.spec\\.ts$",
+    "transform": {
+      "^.+\\.(t|j)s$": "ts-jest"
+    }
   }
 }

--- a/test/index.spec.ts
+++ b/test/index.spec.ts
@@ -1,0 +1,13 @@
+import { DEFAULT_CONFIG } from 'src/config/defaultConfig';
+
+// Temporary test to check alias path working
+it('should pass', () => {
+  const M: CommitMessage = {
+    type: 'feat',
+    scope: 'test',
+    title: 'test',
+    body: 'test',
+  };
+  console.log(M);
+  expect(DEFAULT_CONFIG).toBeDefined();
+});

